libweb3jsonrpc/Debug.cpp

```diff
@@ -142,22 +142,21 @@ Json::Value Debug::debug_storageRangeAt(string const& _blockHashOrNumber, int _t
 
 		// begin is inclusive
 		auto itBegin = storage.lower_bound(h256fromHex(_begin));
-
-		for (auto it = itBegin; it != storage.end(); ++it)
+		auto it = itBegin;
+		for (; it != storage.end(); ++it)
 		{
 			if (ret["storage"].size() == static_cast<unsigned>(_maxResults))
-			{
-				ret["complete"] = false;
 				break;
-			}
 
 			Json::Value keyValue(Json::objectValue);
-			keyValue["hashedKey"] = toCompactHex(it->first, HexPrefix::Add, 1);
+			std::string hashedKey = toCompactHex(it->first, HexPrefix::Add, 1);
 			keyValue["key"] = toCompactHex(it->second.first, HexPrefix::Add, 1);
 			keyValue["value"] = toCompactHex(it->second.second, HexPrefix::Add, 1);
 
-			ret["storage"].append(keyValue);
+			ret["storage"][hashedKey] = keyValue;
 		}
+		if (it != storage.end())
+			ret["nextKey"] = toCompactHex((++it)->first, HexPrefix::Add, 1);
```

Comment:
We could move this into if body inside the loop (before `break`), then we won't need to define `it` outside of loop